src/OpenTelemetry.AutoInstrumentation.StartupHook/StartupHook.cs

```diff
@@ -32,6 +33,17 @@ internal class StartupHook
     /// </summary>
     public static void Initialize()
     {
+        var minSupportedFramework = new FrameworkName(".NETCoreApp,Version=v6.0");
```

Comment:
I think that we should do similiar check on the native code side.

It could be separate PR.